### ISSUE TITLE
[master] Run on single file doesn't allow for all types and performance can be tweaked

### ISSUE THREAD

**Issue by SpacePossum** [FriendsOfPHP/PHP-CS-Fixer#1900]
For example:

```
/home/possum/work/PHP-CS-Fixer$ php php-cs-fixer fix tests/Fixtures/Integration/priority/no_useless_return,no_whitespace_in_blank_lines.test -vvv
```
Causes the iterator to check if the following paths are in fact `tests/Fixtures/Integration/priority/no_useless_return,no_whitespace_in_blank_lines.test`

```
[edit removed after comments]
```

which seems like an overkill since the one file to look for was given.

But more of an issue, if you tell the fixer explicitly to fix the `.test` file it won't do it because of the default Finder settings.

**Comment by keradus** [comment]
1. as it's now building intersection, right?
2. to filter the extension by Finder, the Finder must be used.

**Comment by SpacePossum** [comment]
1. so this is the results because there is a config file at the run location? ok, than it is a UX call, no issue than I guess?
2. can't we tweak the finder to always add the file given by the command to the result? I mean if the user provides it why not fix the file right?

**Comment by keradus** [comment]
pseudocode:
foreach (modifiedFiles as file): php-cs-fixer fix file

so far, if the file is excluded by config then it's not fixed.
after your proposal it would, and that would actually revert whole intersection thing

**Comment by SpacePossum** [comment]
>after your proposal it would, and that would actually revert whole intersection thing

That is not my intention, I understand now, I drop point 1 :)

But how about point 2, if the file is not dropped out by the intersection it can still be filtered out by extension. Would you allow the change that would be, in pseudo;
if the user passes a file by the commandline and it is not filtered by intersect, fix it no matter the extension

**Comment by keradus** [comment]
actually finder filtering different things. Symfony one could filter by path, ext, content, size... 
I wouldn't like to introduce exceptions

**Comment by SpacePossum** [comment]
I think the exceptions should be done on config level for projects like Symfony and not the other way around.
Now if one installs the fixer it is impossible to fix a file called `myFile.test` without changing the extension of the file or by writing a custom Finder in a configuration file.
I'm not asking to change the default behavior to start fixing `.test` files, just if some passes the file by the commandline to honor this and fix the file.

**Comment by keradus** [comment]
Not sure is it that good we should go that way or not, what would you think:

Add new parameter for console run (only for console) like `--path-mode=[override|intersection]`, default is override. Parameter is valid only if any path was provided. Then, `override` is to use passed CLI path as is, and `intersection` to create intersection with finder from config file (if falid, fail if not)

**Comment by keradus** [comment]
easy to do after #1903

**Comment by SpacePossum** [comment]
1903 looks promising, I'll wait for that one :) :+1:
(a master version of https://github.com/FriendsOfPHP/PHP-CS-Fixer/pull/1896 will be way easier as well with 1903 in so +2 from me ;)  )

**Comment by keradus** [comment]
then vote on 1903 ;P

**Comment by keradus** [comment]
If I get you right - your on for path-mode ?

**Comment by SpacePossum** [comment]
yes :) (with default override)  :+1:

**Comment by keradus** [comment]
#1913
